Add code snippet component

bjarnef opened the pull request:
### Prerequisites

- [x] I have added steps to test this contribution in the description below

![2019-12-08_20-32-47](https://user-images.githubusercontent.com/2919859/70394925-071bc400-19fa-11ea-89bf-ce6e30d5220b.gif)

### Description
This PR add a `umb-code-snippet` component, which includes a copy button to copy the code and enhance this feature from this PR https://github.com/umbraco/Umbraco-CMS/pull/6584#issuecomment-549855172

I have been inspired by these code samples https://docs.microsoft.com/en-us/odata/webapi/netcore (which are using highlight.js)

The idea is that package developers also can re-use this component in e.g. the view after installation, in dashboard and in another part of backoffice UI in core.

At the moment this doesn't use third parthy plugins like [highlight.js](https://highlightjs.org/) or [Prism](https://prismjs.com/). https://github.com/umbraco/Umbraco-CMS/issues/7060

Prism is lightweight and it supported by default in TinyMCE, but doesn't out of the box support razor (cshtml/vbhtml)

Highlight.js seems to have some support for razor (cshtml - not sure about vbhtml) https://github.com/highlightjs/highlightjs-cshtml-razor

It could use one of these or maybe the Ace Editor in readonly mode (https://ace.c9.io/#nav=api&api=editor under setReadOnly) https://github.com/Microsoft/monaco-editor/issues/54

It is also possible in Monaco Editor to set readonly mode https://github.com/Microsoft/monaco-editor/issues/54 if Umbraco choose to switch editor https://github.com/umbraco/Umbraco-CMS/pull/6267

Some enhancements that can be added/improved later

- Syntax highlighting: Prism, Highlight.js, Ace Editor, Monaco Editor
- Choose to show/hide copy button or entire "header" (maybe this should be shown by default).

bjarnef commented:
@nielslyngsoe what do you think on this?

nielslyngsoe commented:
Hi @bjarnef 

I think it looks good, but I do feel that we should find a different color than red. It might give the impression that the code is wrong(has errors)?

And now that we are at it, I think it would be nice to use a brighter gray, what do you think?

bjarnef commented:
@nielslyngsoe the red color is how the `<code>` element is styled. Also the gray color is from the `<pre>` element, so I guess it is some general styles that need to be changed?

nielslyngsoe commented:
@bjarnef ah, that originates from Bootstrap CSS Library, which the BackOffice originally was build upon. But now we have decoupled the project so we have our own copy of the project, which we can change as wished. Feel free to change it.

bjarnef commented:
@nielslyngsoe we have this part for styling of `<pre>` element https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/ApiDocs/umbracotemplate/styles/main.css#L201-L211

and this for styling of `<code>` https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Web.UI/Content/bootstrap.css#L1536-L1542 https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Web.UI/Content/bootstrap.css#L1572-L1579

bjarnef commented:
Should it be a neutral text color like this (black or dark gray'ish)?

https://docs.microsoft.com/en-us/odata/webapi/netcore

bjarnef commented:
It might also be possible to re-use the syntax highlighting from ace editor (which already is included):
https://stackoverflow.com/questions/8832320/how-can-i-highlight-code-with-ace-editor

nielslyngsoe commented:
@bjarnef would be interesting to investigate if ACE is a possibility.

If not possible with ACE then I think in terms of color that maybe this one would fit:
@blueExtraDark (#1b264f)
on a background color of:
@brownGrayLight (#f6f4f4)

But feel free to make a choice that you like, you can see color variables in variables.less

bjarnef commented:
@nielslyngsoe I have updated color of `code` element and adjusted the styling a bit.

![image](https://user-images.githubusercontent.com/2919859/71682006-abd9b900-2d8e-11ea-8686-5b8a1837e6ad.png)
